core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt

```diff
@@ -66,10 +65,42 @@ class ArrayPrimitive<T : ValidType>(collection: Collection<TypeExpression<out T>
     },
 )
 
+class ObjectPrimitive(
+    private vararg val entries: ObjectEntry<out ValidType>,
+) : Primitive<ObjectType>(
+    { manager: DopeQueryManager ->
+        val entryDopeQueries = entries.map { it.toDopeQuery(manager) }
+        DopeQuery(
+            queryString = "{${entryDopeQueries.joinToString(", ") { it.queryString }}}",
+            parameters = entryDopeQueries.fold(emptyMap()) { parameters, it -> parameters + it.parameters },
+        )
+    },
+)
+
+fun <K, V> Map<K, V>.toDopeType(): ObjectPrimitive =
+    ObjectPrimitive(
+        *map { (key, value) ->
+            require(key is String) { "Key '$key' is not a String." }
+            key.toDopeType().to(value.toDopeType())
+        }.toTypedArray(),
+    )
+
 fun String.toDopeType() = StringPrimitive(this)
 
 fun Number.toDopeType() = NumberPrimitive(this)
 
 fun Boolean.toDopeType() = BooleanPrimitive(this)
 
 fun <T : ValidType> Collection<TypeExpression<out T>>.toDopeType() = ArrayPrimitive(this)
+
+@JvmName("anyListToDopeType")
+fun <T> Collection<T>.toDopeType(): ArrayPrimitive<ValidType> = map { it.toDopeType() }.toDopeType()
+
+fun <T> T.toDopeType() = when (this) {
```

Comment:
Not exactly, because we dont have the Generic Type yet, so in the `<K, V> Map<K, V>.toDopeType()` function where `V` is Any we can call this funciton.

Reply:
I made it private but I don't think the new name makes much sense, because this function is also used for arrays and up until now we calles all converter functions `toDopeType`.